### PR TITLE
fix(slack): remove integration provider check from proxy logger

### PR DESCRIPTION
**Pull request by cathteng**
Fixes SENTRY-2S2H

Sometimes `self.integration` is not initialized.

### CODE REVIEW

File: src/sentry/api/endpoints/internal/integration_proxy.py [hunk 1]
```diff
@@ -183,6 +185,13 @@ def http_method_not_allowed(self, request):
         self.log_extra["host"] = request.headers.get("Host")
 
         if not self._should_operate(request):
+            self.log_extra["silo_mode"] = SiloMode.get_current_mode().value
```
Comment (ykamo001) [inline src/sentry/api/endpoints/internal/integration_proxy.py:188]: do we still need this?

**Review by ykamo001 — APPROVED** [review]

File: src/sentry/api/endpoints/internal/integration_proxy.py (hunk 1 above)
Comment (cathteng) [reply src/sentry/api/endpoints/internal/integration_proxy.py:188]: probably not, but it doesn't hurt

File: src/sentry/api/endpoints/internal/integration_proxy.py [hunk 2]
```diff
@@ -183,6 +185,13 @@ def http_method_not_allowed(self, request):
...
+            logger.info(
+                "integration_proxy.bad_request",
+                extra={
+                    **self.log_extra,
+                },
```
Comment (ykamo001) [inline src/sentry/api/endpoints/internal/integration_proxy.py:193]: do you need to do **? Looks like everywhere else we're just using the log_extra dict

File: src/sentry/api/endpoints/internal/integration_proxy.py (hunk 1 above)
Comment (ykamo001) [reply src/sentry/api/endpoints/internal/integration_proxy.py:188]: agreed, doesn't hurt

File: src/sentry/api/endpoints/internal/integration_proxy.py (hunk 2 above)
Comment (cathteng) [reply src/sentry/api/endpoints/internal/integration_proxy.py:193]: oh i copied it from when i was passing in some extra stuff

**Pull request merged**